utils/data.py

```
@@ -0,0 +1,34 @@
+    # data 폴더가 없으면 생성
+    if not os.path.exists(DATA_DIR):
+        os.makedirs(DATA_DIR)
+
+    # 파일이 없으면 빈 리스트 저장 후 반환
+    if not os.path.exists(file_path):
+        with open(file_path, 'w', encoding='utf-8') as f:
+            json.dump([], f)
+        return []
+
+    # 파일 읽기
+    with open(file_path, 'r', encoding='utf-8') as f:
```

Ruff (0.14.11)

Unnecessary mode argument

Remove mode argument

(UP015)